pkg/kp/lock.go

```diff
@@ -0,0 +1,100 @@
+package kp
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/square/p2/pkg/util"
+)
+
+type Lock struct {
+	client  *api.Client
+	session string
+	name    string
+}
+
+type AlreadyLockedError struct {
+	Key string
+}
+
+func (err AlreadyLockedError) Error() string {
+	return fmt.Sprintf("Key %q is already locked", err.Key)
+}
+
+func (s *Store) NewLock(name string) (Lock, error) {
```

Comment:
yeah, it basically gives you a thing that you can use to claim locks on kv keys. maybe `Lock` was not the best name. I didn't really like the name `Session` either (which is the consul name for this object)